dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,17 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(val states: Set[State], val start: State, val accept: Set[State], val transitions: Set[Transition]):
+    def accepts(input: String): Boolean = 
+        var currentState = start
+        for (x <- 0 to input.length()-1) {
+            val t = transitions.filter(_.from == currentState).filter(_.symbol == input.charAt(x)).head
```

Comment:
I also did something like this, though I put it in a separate function called "returnTransition" for further clarity. I think this line might warrant something similar, or at least a comment explaining what it does.